shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1191,6 +1206,7 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
             case ComposedType(false, l, r) => Inter(go(l, polarity), go(r, polarity))
             case RecordType(fs) => Record(fs.map(nt => nt._1 -> go(nt._2, polarity)))
             case TupleType(fs) => Tuple(fs.map(nt => nt._1 -> go(nt._2, polarity)))
+            case ArrayType(inner) => Arr(go(inner, polarity)) // ? not sure
```

Comment:
Yeah that's fine because our array types are covariant. (We may later introduce an invariant mutable variation.)
```suggestion
            case ArrayType(inner) => Arr(go(inner, polarity))
```